File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        s1 = 0
        s2 = len(t)
        
        if s2 < len(s):
            return False
        elif len(s) == 0:
            return True
        
        for i in t:
            if s[s1] == i:
                s1 += 1
        
            if s1 == len(s):
                return True
        
        return False

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int s1 = 0;
        int s2 = t.length();
        
        if(t.length() < s.length()){
            return false;
        } else if(s.length() == 0) {
            return true;
        }
        
        
        for(int i = 0; i < s2; i++){
            if(s.charAt(s1) == t.charAt(i)){
                s1++;
            }
            
            if(s1 == s.length()){
                return true;
            }
        }
        
        return false;
    }
}